api/napi/src/interpreter/diagnostic.rs

```diff
@@ -19,12 +23,25 @@ impl From<DiagnosticLevel> for JsDiagnosticLevel {
     }
 }
 
+/// This structure represent a diagnostic emitted while compiling .slint code.
+///
+/// It is basically a message, a level (warning or error), attached to a
+/// position in the code
 #[napi(object, js_name = "Diagnostic")]
 pub struct JsDiagnostic {
+    /// The level for this diagnostic
     pub level: JsDiagnosticLevel,
+
+    /// Message for this diagnostic
     pub message: String,
+
+    /// The line number in the slint source file
```

Comment:
```suggestion
    /// The line number in the Slint source file.
```

(or `.slint` maybe?)